src/GitVersion.Core/VersionCalculation/NextVersionCalculator.cs

```diff
@@ -146,4 +147,177 @@ private static void EnsureHeadIsNotDetached(GitVersionContext context)
+
+        foreach (var effectiveBranchConfiguration in effectiveBranchConfigurationFinder.GetConfigurations(branch, configuration))
+        {
+            foreach (var versionStrategy in this.versionStrategies)
+            {
+                foreach (var baseVersion in versionStrategy.GetBaseVersions(effectiveBranchConfiguration))
+                {
+                    log.Info(baseVersion.ToString());
+                    if (IncludeVersion(baseVersion, configuration.Ignore))
+                    {
+                        var incrementStrategy = incrementStrategyFinder.DetermineIncrementedField(
+                            context: context,
+                            baseVersion: baseVersion,
+                            configuration: effectiveBranchConfiguration.Value
+                        );
+                        var incrementedVersion = incrementStrategy == VersionField.None
+                            ? baseVersion.SemanticVersion
+                            : baseVersion.SemanticVersion.IncrementVersion(incrementStrategy);
+
+                        if (configuration.VersioningMode == VersioningMode.Mainline)
+                        {
+                            if (!(incrementedVersion.PreReleaseTag?.HasTag() != true))
+                            {
+                                continue;
+                            }
+                        }
+
+                        yield return effectiveBranchConfiguration.CreateNextVersion(baseVersion, incrementedVersion);
+                        atLeastOneBaseVersionReturned = true;
+                    }
+                }
+            }
+        }
+
+        if (!atLeastOneBaseVersionReturned)
+        {
+            throw new GitVersionException("No base versions determined on the current branch.");
+        }
+    }
+
+    private bool IncludeVersion(BaseVersion baseVersion, IgnoreConfig ignoreConfiguration)
+    {
+        foreach (var versionFilter in ignoreConfiguration.ToFilters())
+        {
+            if (versionFilter.Exclude(baseVersion, out var reason))
+            {
+                if (reason != null)
+                {
+                    log.Info(reason);
+                }
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private void FixTheBaseVersionSourceOfMergeMessageStrategyIfReleaseBranchWasMergedAndDeleted(IEnumerable<NextVersion> nextVersions)
+    {
+        // TODO: Please us the mechanism per convention and configuration and make the decision in the IVersionStrategy implementation.
+        if (ReleaseBranchExistsInRepo()) return;
+
+        foreach (var nextVersion in nextVersions)
+        {
+            if (nextVersion.BaseVersion.Source.Contains(MergeMessageVersionStrategy.MergeMessageStrategyPrefix)
+                && nextVersion.BaseVersion.Source.Contains("Merge branch")
+                && nextVersion.BaseVersion.Source.Contains("release"))
```

Comment:
Why don't you use the `MergeMessage` class to find a merge message match here?

Reply:
This logic comes from the previous BaseVersionCalculator which I have consolidated with the NextVersionCalculator. I would like to see the method FixTheBaseVersionSourceOfMergeMessageStrategyIfReleaseBranchWasMergedAndDeleted disappearing as soon as possible. Thus I don't want to introduce any bugs because I cannot oversee this logic right now.